hazelcast/src/test/java/com/hazelcast/map/BasicMapTest.java

```diff
@@ -917,6 +918,14 @@ public void testGetPutRemoveAsync() {
         }
     }
 
+    @Test
+    public void testAsyncMethodChaining() {
+        IMap<Integer, Integer> map = getInstance().getMap("testGetPutRemoveAsync");
+        CompletionStage<Integer> setThenGet = map.setAsync(1, 1)
+                                              .thenCompose(v -> map.getAsync(1));
+        assertEquals(1L, (long) setThenGet.toCompletableFuture().join());
```

Comment:
This test would fail with `IllegalStateException` because the response thread that completes the `setAsync` stage is not allowed to make the `getAsync` remote call. Changing the default dependent action executor to `DEFAULT_ASYNC_EXECUTOR` fixes this.